agent/acs/update_handler/updater_test.go

```diff
@@ -435,7 +436,7 @@ func TestNewerUpdateMessages(t *testing.T) {
 
 	require.Equal(t, "newer-update-tar-data", writtenFile.String(), "incorrect data written")
 
-	taskEngine := engine.NewTaskEngine(cfg, nil, nil, nil, nil, nil, nil, nil)
+	taskEngine := engine.NewTaskEngine(cfg, nil, nil, nil, nil, nil, nil, taskresource.ResourceFields{})
```

Comment:
its really weird that this is the only item in this call that isn't a pointer. What's the reasoning here?

Reply:
you're right. This should be a pointer. made necessary changes.